datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.datalake.service.resize.recovery;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
+import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
+
+@Service
+/**
+ * Provides entrypoint for recovery of failed SDX resize.
+ *
+ * Two public methods:
+ * <ul>
+ *     <li>{@code validateRecovery} checks if resize recovery is appropriate</li>
+ *     <li>{@code triggerRecovery} checks if resize recovery is appropriate, and attempts the recovery</li>
+ * </ul>
+ */
+public class ResizeRecoveryService {
+
+    /**
+     * Determines if it's possible to run a <em>Resize</em> recovery.
+     * @return true if resize recovery is possible.
+     */
+    public boolean canRecover() {
+        return false;
+    }
+
+    /**
+     * Determines if it's possible to run a <em>Resize</em> recovery.
+     *
+     * This is distinguished from {@code canRecover} by including a detailed validation message, appropriate as a web response.
+     * @return detailed validation message, as a web response
+     */
+    public SdxRecoverableResponse validateRecovery() {
+        // todo: implement me
+        throw new BadRequestException("SDX Resize Recovery is not yet implemented");
+    }
+
+    public SdxRecoveryResponse triggerRecovery() {
```

Comment:
These methods could be defined in a generic recovery service interface. along with the same parameters as sdxUpgradeRecoveryService

Reply:
I've played around a bit with this and I'm not sure I have a diverse enough implementation to warrant extracting an interface yet. We've got 3 things that look like they have the same functionality, but one of them just delegates to one of the other two:
* `RecoveryService` delegates to `ResizeRecoveryService` or `SdxUpgradeRecoveryService`
* `ResizeRecoveryService` is not yet implemented
* `SdxUpgradeRecoveryService` is in flux, I'm not sure how far along it's implementation is

I think to properly interface this too, we'll need to define a `RecoveryResponse<SdxCluster>` or `RecoveryResponse<Datalake>`.

And I did some other renaming extraction to try to get things to line up.

Check out a preview of it [here](https://github.com/hortonworks/cloudbreak/pull/11709)

Reply:
Talked to Dan, we'll review the [second PR](https://github.com/hortonworks/cloudbreak/pull/11709) to make the interface change.